docs/book/features/vsphere_csi_migration.md

```diff
@@ -0,0 +1,175 @@
+              certificatesigningrequest.certificates.k8s.io/vsphere-webhook-svc.kube-system created
+              NAME                              AGE   SIGNERNAME                     REQUESTOR          CONDITION
+              vsphere-webhook-svc.kube-system   0s    kubernetes.io/legacy-unknown   kubernetes-admin   Pending
+              certificatesigningrequest.certificates.k8s.io/vsphere-webhook-svc.kube-system approved
+              secret/vsphere-webhook-certs configured
+
+     2. Create ValidatingWebhookConfiguration, WebHook Deployment Pod, Service Accounts, Cluster Role,  Role Bindings, Service to bind with webhook pod.
+
+            $ cd vsphere-csi-driver/manifests/v2.1.0/vsphere-7.0u1/vanilla/deploy/
+            $ ./create-validation-webhook.sh
+            service/vsphere-webhook-svc created
+            validatingwebhookconfiguration.admissionregistration.k8s.io/validation.csi.vsphere.vmware.com created
+            serviceaccount/vsphere-csi-webhook created
+            clusterrole.rbac.authorization.k8s.io/vsphere-csi-webhook-role created
+            clusterrolebinding.rbac.authorization.k8s.io/vsphere-csi-webhook-role-binding created
+            deployment.apps/vsphere-csi-webhook created
+
+5. Enable feature flags `CSIMigration` and `CSIMigrationvSphere` on `kube-controller` and `kubelet` on all controllers and workload nodes. `CSIMigrationvSphere` flag enables shims and translation logic to route volume operations from the vSphere in-tree plugin to vSphere CSI plugin. Supports falling back to in-tree vSphere plugin if a node does not have a vSphere CSI plugin installed and configured. This flag requires `CSIMigration` feature flag to be enabled.
+
+   - update kube-controller-manager manifest file and following arguments. This file is generally available at `/etc/kubernetes/manifests/kube-controller-manager.yaml`
+
+        `- --feature-gates=CSIMigration=true,CSIMigrationvSphere=true`
+
+   - update kubelet configuration file and add following flags. This file is generally available at `/var/lib/kubelet/config.yaml`
+
+          featureGates:
+          {
+            "CSIMigration": true,
+            "CSIMigrationvSphere": true
+          }
+
+      Please note that before changing the configuration on the Kubelet on each node we **must drain** the node (remove running workloads).
+
+      There is also an optional `CSIMigrationvSphereComplete` flag that can be enabled if all the nodes have CSI migration enabled. `CSIMigrationvSphereComplete` helps stop registering the vSphere in-tree plugin in kubelet and volume controllers and enables shims and translation logic to route volume operations from the vSphere in-tree plugin to vSphere CSI plugin. Requires `CSIMigration` and `CSIMigrationvSphere` feature flags enabled and vSphere CSI plugin installed and configured on all nodes in the cluster.
```

Comment:
May be we should add a section on what to do to switchover/complete migration and move this block there.

Reply:
moved to step-6